Source/Core/AudioCommon/Mixer.h

```diff
@@ -4,51 +4,70 @@
 
 #pragma once
 
+#include <array>
 #include <atomic>
 #include <mutex>
 #include <string>
+#include <vector>
 
 #include "AudioCommon/WaveFile.h"
 
-// 16 bit Stereo
-#define MAX_SAMPLES     (1024 * 2) // 64ms
-#define INDEX_MASK      (MAX_SAMPLES * 2 - 1)
-
-#define LOW_WATERMARK   1280 // 40 ms
-#define MAX_FREQ_SHIFT  200  // per 32000 Hz
-#define CONTROL_FACTOR  0.2f // in freq_shift per fifo size offset
-#define CONTROL_AVG     32
+// Produces white noise in the range of [-4.f, 4.f]
+// Note: for me this values produces more natural results than master [-0.5, 0.5]
+const float rndrcp = 8.f / float(RAND_MAX);
+#define DITHER_NOISE ((rand() * rndrcp) - 4.f)
+
+// converts [-32768, 32767] -> [-1.0, 1.0)
+constexpr float Signed16ToFloat(const s16 s)
+{
+	return s * 0.000030517578125f;
+}
+
+// we NEED dithering going from float -> 16bit
+__forceinline void TriangleDither(float& sample, float& prev_dither)
+{
+	float dither = DITHER_NOISE;
```

Comment:
Crap, random dithering? This will call rand() per sample (with DPL, even more often).

Please be kindful and just use correct rounding with an error integral. Or just correct rounding ...